src/EA.Weee.Web/Areas/Admin/Controllers/ReportsController.cs

```diff
@@ -477,8 +477,8 @@ public async Task<ActionResult> AatfAeReturnData(AatfAeReturnDataViewModel model
           int quarter,  FacilityType facilityType, int? submissionStatus, Guid? authority, Guid? pat, Guid? localArea)
         {
             CSVFileData fileData;
-
-            GetAatfAeReturnDataCsv request = new GetAatfAeReturnDataCsv(complianceYear, quarter, facilityType, submissionStatus, authority, pat, localArea);
+            string aatfDataUrl = HttpContext.Request.Url != null ? string.Concat(HttpContext.Request.Url.Scheme, "://", HttpContext.Request.Url.Authority, "/admin/aatf/details/") : string.Empty;
```

Comment:
We should be able to mock the HTTP context using HttpContextMocker so you can test that the request is passed the the correct URL based on the mocked httpcontext

Reply:
Added